promql/functions.go

```diff
@@ -570,6 +570,70 @@ func funcLog10(vals []parser.Value, args parser.Expressions, enh *EvalNodeHelper
 	return simpleFunc(vals, enh, math.Log10)
 }
 
+// === sin(Vector parser.ValueTypeVector) Vector ===
+func funcSin(vals []parser.Value, args parser.Expressions, enh *EvalNodeHelper) Vector {
+	return simpleFunc(vals, enh, math.Sin)
+}
+
+// === cos(Vector parser.ValueTypeVector) Vector ===
+func funcCos(vals []parser.Value, args parser.Expressions, enh *EvalNodeHelper) Vector {
+	return simpleFunc(vals, enh, math.Cos)
+}
+
+// === tan(Vector parser.ValueTypeVector) Vector ===
+func funcTan(vals []parser.Value, args parser.Expressions, enh *EvalNodeHelper) Vector {
+	return simpleFunc(vals, enh, math.Tan)
+}
+
+// == asin(Vector parser.ValueTypeVector) Vector ===
+func funcAsin(vals []parser.Value, args parser.Expressions, enh *EvalNodeHelper) Vector {
+	return simpleFunc(vals, enh, math.Asin)
+}
+
+// == acos(Vector parser.ValueTypeVector) Vector ===
+func funcAcos(vals []parser.Value, args parser.Expressions, enh *EvalNodeHelper) Vector {
+	return simpleFunc(vals, enh, math.Acos)
+}
+
+// == atan(Vector parser.ValueTypeVector) Vector ===
+func funcAtan(vals []parser.Value, args parser.Expressions, enh *EvalNodeHelper) Vector {
+	return simpleFunc(vals, enh, math.Atan)
+}
+
+// === atan2(Vector1, Vector2 parser.ValueTypeVector) Vector ===
+func funcAtan2(vals []parser.Value, args parser.Expressions, enh *EvalNodeHelper) Vector {
```

Comment:
I do not think we ever have a function that takes 2 vectors.

Reply:
Yes, but there's not really any other way to implement `atan2`.

In the example provided by @leonerd:

> Having done that, your longerterm averages over data_x and data_y now make sense, and you can then extract an "average" magnitude and direction out of these again:
```
data_r:avg1d = sqrt(data_x:avg1d ^ 2 + data_y:avg1d ^ 2)
data_theta:avg1d = atan2(data_y:avg1d, data_x:avg1d)
```

I don't think it's meant to use the UNIX time and as the x.